### PR TITLE
Add Follow

### LINKED ISSUE #34: DEBUG is always True

### LINKED ISSUE #37: Add Follow system
Users will get activities of following users as notifications.

### PR DESCRIPTION
**Pull request by AmirSavand**
Resolves #37 
Fixes #34 

- [x] Follow model
- [x] User followings API
- [x] User followers API
- [x] User followings count
- [x] User followers count
- [x] Return follow id
- [x] Unfollow

### CODE REVIEW

**Comment by AmirSavand** [comment]
@Alir3z4 ready to review.

**Comment by AmirSavand** [comment]
@Alir3z4 ready to review :rocket:

File: account/models.py [hunk 1]
```diff
@@ -86,3 +87,22 @@ class Meta:
...
         ordering = ['-id']
+
+
+class Follow(models.Model):
+    user = models.ForeignKey(User, related_name='following', on_delete=models.CASCADE)
+    following = models.ForeignKey(User, related_name='follower', on_delete=models.CASCADE)
+
+    def save(self, *args, **kwargs):
+        if self.user == self.following:
+            raise ValidationError("Users cannot follow themselves.")
```
Comment (Alir3z4) [inline account/models.py]: You should not apply that in here.
This should be taken care of in an API or another method.

File: api/serializers.py [hunk 2]
```diff
@@ -63,6 +80,53 @@ class Meta:
         ]
 
 
+class FollowSerializer(serializers.ModelSerializer):
+    user = UserMinimalSerializer(read_only=True)
+    following = UserMinimalSerializer(read_only=True)
+
+    class Meta:
+        model = Follow
+        fields = [
```
Comment (Alir3z4) [inline api/serializers.py]: You may use a tuple `()` instead of list `[]`

File: account/models.py (hunk 1 above)
Comment (AmirSavand) [reply account/models.py]: @Alir3z4 so I delete the `save` method completely?

**Comment by AmirSavand** [comment]
@Alir3z4 changes done, review again.

**Pull request merged**